src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/UsageBasedMetadataManager.cs

```diff
@@ -783,15 +783,28 @@ public bool GeneratesAttributeMetadata(TypeDesc attributeType)
 
         public override void NoteOverridingMethod(MethodDesc baseMethod, MethodDesc overridingMethod)
         {
-            // We validate that the various dataflow/Requires* annotations are consistent across virtual method overrides
-            if (HasMismatchingAttributes(baseMethod, overridingMethod, "RequiresUnreferencedCodeAttribute"))
-            {
-                Logger.LogWarning(overridingMethod, DiagnosticId.RequiresUnreferencedCodeAttributeMismatch, overridingMethod.GetDisplayName(), baseMethod.GetDisplayName());
-            }
+            bool baseMethodTypeIsInterface = baseMethod.OwningType.IsInterface;
+            string overridingMethodName = overridingMethod.GetDisplayName();
+            string baseMethodName = baseMethod.GetDisplayName();
```

Comment:
We should compute these when needed. For 99.9% of calls to this method, these strings won't be needed.